target/linux/ramips/mt76x8/base-files/etc/board.d/02_network

```diff
@@ -181,6 +182,9 @@ ramips_setup_macs()
 		wan_mac=$(mtd_get_mac_binary factory 0x4)
 		lan_mac=$(mtd_get_mac_binary factory 0x2e)
 		;;
+	netgear,r6020)
+		wan_mac=$(macaddr_add "$(mtd_get_mac_binary factory 0x8004)" -4)
```

Comment:
"-4" seems rather unprobable to me. Can you please report the MAC address assignment on vendor firmware? I'd need the last byte for LAN, WAN, 2.4g and 5g interfaces ...

Reply:
I'm wondering if this was meant to be the mac address for a wireless device?  I think I might have a partially corrupted config partition.

Reply:
Well, the art of copying together code from other similar devices is to find out what to keep, what to remove, and what to adjust properly.
The MAC address assignment is one of the latter kind. And to adjust them properly, you first have to find out what you have to adjust them to. So, please, flash back to vendor firmware, check the MAC addresses there as requested above, and then report them here. From this point, we may move forward towards how to implement them properly.

Further reading:
https://openwrt.org/docs/guide-developer/mac.address

Reply:
Did that.  See my other comment.